test/repository-test.js

```diff
@@ -0,0 +1,101 @@
+const chai = require('chai');
+const expect = chai.expect;
+
+import Repository from '../src/Repository';
+import User from '../src/User';
+import Sleep from '../src/Sleep-Repository';
+import Hydration from '../src/Hydration-Repository'
+import Activity from '../src/Activity-Repository';
+import userTestData from '../test-data/user-test-data';
+import sleepTestData from '../test-data/sleep-test-data';
+import hydroTestData from '../test-data/hydration-test-data';
+import activityTestData from '../test-data/activity-test-data';
+
+describe('Repository', () => {
+  let userRepo, user, sleepRepo, hydroRepo, activityRepo;
+  beforeEach(() => {
+    userRepo = new Repository(userTestData);
+    user = new User(userRepo.findUser(1));
+    sleepRepo = new Sleep(sleepTestData);
+    hydroRepo = new Hydration(hydroTestData);
+    activityRepo = new Activity(activityTestData);
+  });
+
+  it("should be able to locate a user from the dataset", () => {
+    expect(userRepo.findUser(1)).to.deep.equal({
+      "id": 1,
+      "name": "Luisa Hane",
+      "address": "15195 Nakia Tunnel, Erdmanport VA 19901-1697",
+      "email": "Diana.Hayes1@hotmail.com",
+      "strideLength": 4.3,
+      "dailyStepGoal": 10000,
+      "friends": [
+        16,
+        4,
+        8
+      ]
+    });
+  });
+
+  describe('findToday method', () => {
+    it("should be able to find the most current data point", () => {
+      expect(hydroRepo.findToday(hydroRepo.data)).to.equal("2019/06/23");
+      expect(sleepRepo.findToday(sleepRepo.data)).to.equal("2019/06/22");
+      expect(activityRepo.findToday(activityRepo.data)).to.equal("2019/06/21");
```

Comment:
@JamesRexMiller4 Your testing structure is getting a little conflated here. Even though these methods are inherited from the parent class and invoked on the subclasses, I would still unit test them from the parent class itself. Since this test file is called `repository-test`, it's confusing to see subclasses being tested.

Reply:
@khalidwilliams are you saying I should move my tests, for the methods that are inherited, into the child test files? Or that I should create a describe block within the repository test file that describes the instance of the child, and then test all the methods that pertain to it?

Our repository class is our parent class, and we instantiate our userRepository as an instance of that parent while the other classes instantiate as children of the repository

Reply:
describe("Activity.findMetricByDate") vs describe( findMetricByDate method for all classes)